frontend/src/metabase/visualizations/components/settings/ChartSettingFieldsPicker.jsx

```diff
@@ -10,62 +11,106 @@ const ChartSettingFieldsPicker = ({
-            } else {
-              // otherwise leave it blank
-              onChange(value.concat([undefined]));
-            }
-          }}
-        >
-          {addAnother}
-        </a>
-      </AddAnotherContainer>
-    )}
-  </div>
-);
+}) => {
+  const handleDragEnd = ({ source, destination }) => {
+    const oldIndex = source.index,
+      newIndex = destination.index;
+
+    const valueCopy = [...value];
+    valueCopy.splice(newIndex, 0, valueCopy.splice(oldIndex, 1)[0]);
+    onChange(valueCopy);
+  };
+
+  const calculateOptions = item => {
+    return options.filter(
+      option =>
+        value.findIndex(v => v === option.value) < 0 || option.value === item,
+    );
+  };
+
+  return (
+    <div>
+      {Array.isArray(value) ? (
+        <DragDropContext onDragEnd={handleDragEnd}>
+          <Droppable droppableId="droppable">
+            {provided => (
+              <div {...provided.droppableProps} ref={provided.innerRef}>
+                {value.map((v, index) => {
+                  return (
+                    <Draggable
+                      key={`draggable-${v}`}
+                      draggableId={`draggable-${v}`}
+                      index={index}
+                    >
+                      {provided => (
+                        <div
+                          ref={provided.innerRef}
+                          {...provided.draggableProps}
+                          {...provided.dragHandleProps}
+                          className="mb1"
+                        >
+                          <ChartSettingFieldPicker
+                            {...props}
+                            key={index}
+                            value={v}
+                            options={calculateOptions(v)}
+                            onChange={v => {
+                              const newValue = [...value];
+                              // this swaps the position of the existing value
+                              const existingIndex = value.indexOf(v);
+                              if (existingIndex >= 0) {
+                                newValue.splice(existingIndex, 1, value[index]);
+                              }
+                              // replace with the new value
+                              newValue.splice(index, 1, v);
+                              onChange(newValue);
+                            }}
+                            onRemove={
+                              value.filter(v => v != null).length > 1 ||
+                              (value.length > 1 && v == null)
+                                ? () =>
+                                    onChange([
+                                      ...value.slice(0, index),
+                                      ...value.slice(index + 1),
+                                    ])
+                                : null
+                            }
+                            dragHandle={value.length > 1}
```

Comment:
nit: It isn't clear if a `dragHandle` is a boolean, if I saw this inside a `<ChartSettingFieldPicker />`  component then I would have thought this is a React element that renders the drag handle.

We seems to have an established pattern of the name `showSomething` as a boolean, so following that this variable could be named `showDragHandle`. What do you think?